### PR TITLE
fix(components): [el-form] if required is not true don't validate

### PR DESCRIPTION
**Pull request by btea**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Review by ryuhangyeong — APPROVED** [review]

**Comment by tolking** [comment]
I think there is a problem with this change.

step:

click the `Submit` button

[before](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtZm9ybVxuICAgIHJlZj1cImZvcm1SZWZcIlxuICAgIDptb2RlbD1cIm51bWJlclZhbGlkYXRlRm9ybVwiXG4gICAgbGFiZWwtd2lkdGg9XCIxMDBweFwiXG4gICAgY2xhc3M9XCJkZW1vLXJ1bGVGb3JtXCJcbiAgPlxuICAgIDxlbC1mb3JtLWl0ZW1cbiAgICAgIGxhYmVsPVwiYWdlXCJcbiAgICAgIHByb3A9XCJhZ2VcIlxuICAgICAgOnJ1bGVzPVwiW1xuICAgICAgICB7IHJlcXVpcmVkOiB0cnVlLCBtZXNzYWdlOiAnYWdlIGlzIHJlcXVpcmVkJyB9LFxuICAgICAgICB7IHR5cGU6ICdudW1iZXInLCBtZXNzYWdlOiAnYWdlIG11c3QgYmUgYSBudW1iZXInIH0sXG4gICAgICBdXCJcbiAgICA+XG4gICAgICA8ZWwtaW5wdXRcbiAgICAgICAgdi1tb2RlbC5udW1iZXI9XCJudW1iZXJWYWxpZGF0ZUZvcm0uYWdlXCJcbiAgICAgICAgdHlwZT1cInRleHRcIlxuICAgICAgICBhdXRvY29tcGxldGU9XCJvZmZcIlxuICAgICAgLz5cbiAgICA8L2VsLWZvcm0taXRlbT5cbiAgICA8ZWwtZm9ybS1pdGVtPlxuICAgICAgPGVsLWJ1dHRvbiB0eXBlPVwicHJpbWFyeVwiIEBjbGljaz1cInN1Ym1pdEZvcm0oZm9ybVJlZilcIj5TdWJtaXQ8L2VsLWJ1dHRvbj5cbiAgICAgIDxlbC1idXR0b24gQGNsaWNrPVwicmVzZXRGb3JtKGZvcm1SZWYpXCI+UmVzZXQ8L2VsLWJ1dHRvbj5cbiAgICA8L2VsLWZvcm0taXRlbT5cbiAgPC9lbC1mb3JtPlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmltcG9ydCB7IHJlYWN0aXZlLCByZWYgfSBmcm9tICd2dWUnXG5pbXBvcnQgdHlwZSB7IEZvcm1JbnN0YW5jZSB9IGZyb20gJ2VsZW1lbnQtcGx1cydcblxuY29uc3QgZm9ybVJlZiA9IHJlZjxGb3JtSW5zdGFuY2U+KClcblxuY29uc3QgbnVtYmVyVmFsaWRhdGVGb3JtID0gcmVhY3RpdmUoe1xuICBhZ2U6ICdmZicsXG59KVxuXG5jb25zdCBzdWJtaXRGb3JtID0gKGZvcm1FbDogRm9ybUluc3RhbmNlIHwgdW5kZWZpbmVkKSA9PiB7XG4gIGlmICghZm9ybUVsKSByZXR1cm5cbiAgZm9ybUVsLnZhbGlkYXRlKCh2YWxpZCkgPT4ge1xuICAgIGlmICh2YWxpZCkge1xuICAgICAgY29uc29sZS5sb2coJ3N1Ym1pdCEnKVxuICAgIH0gZWxzZSB7XG4gICAgICBjb25zb2xlLmxvZygnZXJyb3Igc3VibWl0IScpXG4gICAgICByZXR1cm4gZmFsc2VcbiAgICB9XG4gIH0pXG59XG5cbmNvbnN0IHJlc2V0Rm9ybSA9IChmb3JtRWw6IEZvcm1JbnN0YW5jZSB8IHVuZGVmaW5lZCkgPT4ge1xuICBpZiAoIWZvcm1FbCkgcmV0dXJuXG4gIGZvcm1FbC5yZXNldEZpZWxkcygpXG59XG48L3NjcmlwdD5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiX28iOnt9fQ==)

[after](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtZm9ybVxuICAgIHJlZj1cImZvcm1SZWZcIlxuICAgIDptb2RlbD1cIm51bWJlclZhbGlkYXRlRm9ybVwiXG4gICAgbGFiZWwtd2lkdGg9XCIxMDBweFwiXG4gICAgY2xhc3M9XCJkZW1vLXJ1bGVGb3JtXCJcbiAgPlxuICAgIDxlbC1mb3JtLWl0ZW1cbiAgICAgIGxhYmVsPVwiYWdlXCJcbiAgICAgIHByb3A9XCJhZ2VcIlxuICAgICAgOnJ1bGVzPVwiW1xuICAgICAgICB7IHJlcXVpcmVkOiB0cnVlLCBtZXNzYWdlOiAnYWdlIGlzIHJlcXVpcmVkJyB9LFxuICAgICAgICB7IHR5cGU6ICdudW1iZXInLCBtZXNzYWdlOiAnYWdlIG11c3QgYmUgYSBudW1iZXInIH0sXG4gICAgICBdXCJcbiAgICA+XG4gICAgICA8ZWwtaW5wdXRcbiAgICAgICAgdi1tb2RlbC5udW1iZXI9XCJudW1iZXJWYWxpZGF0ZUZvcm0uYWdlXCJcbiAgICAgICAgdHlwZT1cInRleHRcIlxuICAgICAgICBhdXRvY29tcGxldGU9XCJvZmZcIlxuICAgICAgLz5cbiAgICA8L2VsLWZvcm0taXRlbT5cbiAgICA8ZWwtZm9ybS1pdGVtPlxuICAgICAgPGVsLWJ1dHRvbiB0eXBlPVwicHJpbWFyeVwiIEBjbGljaz1cInN1Ym1pdEZvcm0oZm9ybVJlZilcIj5TdWJtaXQ8L2VsLWJ1dHRvbj5cbiAgICAgIDxlbC1idXR0b24gQGNsaWNrPVwicmVzZXRGb3JtKGZvcm1SZWYpXCI+UmVzZXQ8L2VsLWJ1dHRvbj5cbiAgICA8L2VsLWZvcm0taXRlbT5cbiAgPC9lbC1mb3JtPlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmltcG9ydCB7IHJlYWN0aXZlLCByZWYgfSBmcm9tICd2dWUnXG5pbXBvcnQgdHlwZSB7IEZvcm1JbnN0YW5jZSB9IGZyb20gJ2VsZW1lbnQtcGx1cydcblxuY29uc3QgZm9ybVJlZiA9IHJlZjxGb3JtSW5zdGFuY2U+KClcblxuY29uc3QgbnVtYmVyVmFsaWRhdGVGb3JtID0gcmVhY3RpdmUoe1xuICBhZ2U6ICdmZicsXG59KVxuXG5jb25zdCBzdWJtaXRGb3JtID0gKGZvcm1FbDogRm9ybUluc3RhbmNlIHwgdW5kZWZpbmVkKSA9PiB7XG4gIGlmICghZm9ybUVsKSByZXR1cm5cbiAgZm9ybUVsLnZhbGlkYXRlKCh2YWxpZCkgPT4ge1xuICAgIGlmICh2YWxpZCkge1xuICAgICAgY29uc29sZS5sb2coJ3N1Ym1pdCEnKVxuICAgIH0gZWxzZSB7XG4gICAgICBjb25zb2xlLmxvZygnZXJyb3Igc3VibWl0IScpXG4gICAgICByZXR1cm4gZmFsc2VcbiAgICB9XG4gIH0pXG59XG5cbmNvbnN0IHJlc2V0Rm9ybSA9IChmb3JtRWw6IEZvcm1JbnN0YW5jZSB8IHVuZGVmaW5lZCkgPT4ge1xuICBpZiAoIWZvcm1FbCkgcmV0dXJuXG4gIGZvcm1FbC5yZXNldEZpZWxkcygpXG59XG48L3NjcmlwdD5cbiIsIlBsYXlncm91bmRNYWluLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgQXBwIGZyb20gJy4vQXBwLnZ1ZSdcbmltcG9ydCB7IHNldHVwRWxlbWVudFBsdXMgfSBmcm9tICcuL2VsZW1lbnQtcGx1cy5qcydcbnNldHVwRWxlbWVudFBsdXMoKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPEFwcCAvPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcImVsZW1lbnQtcGx1c1wiOiBcImh0dHBzOi8vcHJldmlldy0xMTE4OC1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmZ1bGwubWluLm1qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzL1wiOiBcInVuc3VwcG9ydGVkXCJcbiAgfVxufSIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcInZ1ZVwiOiBcImh0dHBzOi8vdW5wa2cuY29tL0B2dWUvcnVudGltZS1kb21AbGF0ZXN0L2Rpc3QvcnVudGltZS1kb20uZXNtLWJyb3dzZXIuanNcIixcbiAgICBcIkB2dWUvc2hhcmVkXCI6IFwiaHR0cHM6Ly91bnBrZy5jb20vQHZ1ZS9zaGFyZWRAbGF0ZXN0L2Rpc3Qvc2hhcmVkLmVzbS1idW5kbGVyLmpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXNcIjogXCJodHRwczovL3ByZXZpZXctMTExODgtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiLFxuICAgIFwiQGVsZW1lbnQtcGx1cy9pY29ucy12dWVcIjogXCJodHRwczovL3VucGtnLmNvbS9AZWxlbWVudC1wbHVzL2ljb25zLXZ1ZUAyL2Rpc3QvaW5kZXgubWluLmpzXCJcbiAgfSxcbiAgXCJzY29wZXNcIjoge31cbn0iLCJlbGVtZW50LXBsdXMuanMiOiJpbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5pbXBvcnQgRWxlbWVudFBsdXMgZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5sZXQgaW5zdGFsbGVkID0gZmFsc2VcbmF3YWl0IGxvYWRTdHlsZSgpXG5cbmV4cG9ydCBmdW5jdGlvbiBzZXR1cEVsZW1lbnRQbHVzKCkge1xuICBpZiAoaW5zdGFsbGVkKSByZXR1cm5cbiAgY29uc3QgaW5zdGFuY2UgPSBnZXRDdXJyZW50SW5zdGFuY2UoKVxuICBpbnN0YW5jZS5hcHBDb250ZXh0LmFwcC51c2UoRWxlbWVudFBsdXMpXG4gIGluc3RhbGxlZCA9IHRydWVcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGxvYWRTdHlsZSgpIHtcbiAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICBjb25zdCBsaW5rID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnbGluaycpXG4gICAgbGluay5yZWwgPSAnc3R5bGVzaGVldCdcbiAgICBsaW5rLmhyZWYgPSAnaHR0cHM6Ly9wcmV2aWV3LTExMTg4LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzJ1xuICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignbG9hZCcsIHJlc29sdmUpXG4gICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdlcnJvcicsIHJlamVjdClcbiAgICBkb2N1bWVudC5ib2R5LmFwcGVuZChsaW5rKVxuICB9KVxufSIsIl9vIjp7InNob3dIaWRkZW4iOnRydWUsInN0eWxlU291cmNlIjoiaHR0cHM6Ly9wcmV2aWV3LTExMTg4LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzIn19)

**Comment by tolking** [comment]
If the `required` is set to `false`, it is still inconsistent with before.

**Comment by btea** [comment]
I think required is not set or set to `false`, there should be no need to validate.

**Comment by tolking** [comment]
Link #9618

File: packages/components/form/src/form-item.vue [hunk 1]
```diff
@@ -303,7 +303,7 @@ const validate: FormItemContext['validate'] = async (trigger, callback) => {
   }
 
   const rules = getFilteredRule(trigger)
-  if (rules.length === 0) {
+  if (rules.length === 0 || rules.every((rule) => !rule.required)) {
```
Comment (jw-foss) [inline packages/components/form/src/form-item.vue:306]: **Nitpick:** perhaps we could do:

```suggestion
  if (rules.length === 0 || !rules.some((rule) => rule.required)) {
```

If any rule contains `required == true`, then this expression should be false thus the condition won't hold. WDTY?

**Comment by jw-foss** [comment]
@btea I think you have some misunderstanding on the rule validation.

`required` is a field that is set to determine whether the field should be presented or not:
1. When it is true, the field must be presented.
2. And the rule is how we want to validate it, when the `required` is not true, but the field is being presented in the form, we should validate them.

Combining the conditions above, we should say that `required` has nothing to do with `rule` itself.

So I think this is not really what we want, if we merge this code, probably hundreds of thousands users will be impacted.

**Comment by btea** [comment]
@jw-foss Thank you very much for the detailed explanation, it seems that I have been misunderstanding this field before.

File: packages/components/form/src/form-item.vue (hunk 1 above)
Comment (btea) [reply packages/components/form/src/form-item.vue:306]: According to the second point mentioned below, if required is set to false, it should also be validate. So the judgment here may not be necessary, we should close this pr.

**Pull request closed without merging**